realm/src/androidTest/java/io/realm/RealmCursorTest.java

```diff
@@ -0,0 +1,595 @@
+        }
+    }
+
+    public void testCopyStringToBuffer() {
+        String expectedString = "test data 0";
+        int expectedLength = expectedString.length();
+        cursor.moveToFirst();
+        CharArrayBuffer buffer = new CharArrayBuffer(expectedLength);
+        cursor.copyStringToBuffer(AllTypes.COL_INDEX_STRING, buffer);
+        assertEquals(expectedLength, buffer.sizeCopied);
+        assertEquals(expectedLength, buffer.data.length);
+        assertEquals("test data 0", new String(buffer.data));
+    }
+
+    public void testGetShort() {
+        cursor.moveToFirst();
+        short value = cursor.getShort(AllTypes.COL_INDEX_LONG);
+        assertEquals(0, value);
+    }
+
+    public void testGetInt() {
+        cursor.moveToFirst();
+        int value = cursor.getInt(AllTypes.COL_INDEX_LONG);
+        assertEquals(0, value);
+    }
+
+    public void testGetLong() {
+        cursor.moveToFirst();
+        long value = cursor.getLong(AllTypes.COL_INDEX_LONG);
+        assertEquals(0, value);
+    }
+
+    public void testGetFloat() {
+        cursor.moveToFirst();
+        float value = cursor.getFloat(AllTypes.COL_INDEX_FLOAT);
+        assertEquals(1.234567f, value);
+    }
+
+    public void testGetDouble() {
+        cursor.moveToFirst();
+        double value = cursor.getDouble(AllTypes.COL_INDEX_DOUBLE);
+        assertEquals(3.1415d, value);
+    }
+
+    public void testGetTypeInvalidIndexThrows() {
+        try {
+            cursor.getType(-1);
+            fail();
+        } catch (IndexOutOfBoundsException expected) {
+        }
+    }
+
+    public void testGetType() {
+        assertEquals(Cursor.FIELD_TYPE_STRING, cursor.getType(AllTypes.COL_INDEX_STRING));
+        assertEquals(Cursor.FIELD_TYPE_INTEGER, cursor.getType(AllTypes.COL_INDEX_LONG));
+        assertEquals(Cursor.FIELD_TYPE_FLOAT, cursor.getType(AllTypes.COL_INDEX_FLOAT));
+        assertEquals(Cursor.FIELD_TYPE_FLOAT, cursor.getType(AllTypes.COL_INDEX_DOUBLE));
+        assertEquals(Cursor.FIELD_TYPE_INTEGER, cursor.getType(AllTypes.COL_INDEX_BOOLEAN));
+        assertEquals(Cursor.FIELD_TYPE_INTEGER, cursor.getType(AllTypes.COL_INDEX_DATE));
+        assertEquals(-1, cursor.getType(AllTypes.COL_INDEX_OBJECT));
+        assertEquals(-1, cursor.getType(AllTypes.COL_INDEX_LIST));
+    }
+
+    public void testIsNullThrows() {
+        cursor.moveToFirst();
+        try {
+            cursor.isNull(AllTypes.COL_INDEX_STRING);
+            fail();
+        } catch (UnsupportedOperationException expected) {
+        }
+    }
+
+    public void testDeactivateThrows() {
+        try {
+            cursor.deactivate();
+            fail();
+        } catch (UnsupportedOperationException expected) {
+        }
+    }
+
+    public void testRequeryThrows() {
+        try {
+            cursor.requery();
+            fail();
+        } catch (UnsupportedOperationException expected) {
+        }
+    }
+
+    public void testClose() {
```

Comment:
Test order could be a bit better. testClose() seems unrelated here. Collect the tests for "unsupported" methods.

I would personally have collected them all in one method. Literally compressing lines by a factor of 10 down to less than 10 lines total - completely disregarding normal structure and max line size :-) e.g.:
```
try { cursor.requery();       fail(); } catch (UnsupportedOperationException expected) {} }
try { cursor.deactivate();    fail(); } catch (UnsupportedOperationException expected) {} }
// etc..
```
Makes it easier to overview which methods to are unsupported.